src/Repository/SubmitRepository.php

```diff
@@ -54,6 +54,60 @@ public function findUserSubmits(User $user): array
         ;
     }
 
+    public function updateStatuses(): void
+    {
+        $em = $this->getEntityManager();
+
+        foreach ($this->getDoneSubmits() as $submit) {
+            $submit->setStatus(Submit::STATUS_DONE);
+            $submit->resetStatusChanged();
+        }
+
+        foreach ($this->getOldPendingSubmits() as $submit) {
+            $em->remove($submit);
```

Comment:
Do we really want to **remove** things?

Anyway, if yes, **one** `delete` SQL query would be much more performant and a SELECT + many DELETE